File: src/app/model/produit.ts
export class Produit {
  id: number;
  nom: string;
  prix: number;
  categoryId: number; // ID de la catégorie
  categoryName: string; // Nom de la catégorie

  constructor(
    id: number,
    nom: string,
    prix: number,
    categoryId: number,
    categoryName: string
  ) {
    this.id = id;
    this.nom = nom;
    this.prix = prix;
    this.categoryId = categoryId;
    this.categoryName = categoryName;
  }
}

File: backEnd/src/main/java/com/example/Entity/Produit.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.example.Entity;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import java.io.*;
import javax.persistence.*;

@Entity
public class Produit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nom;

    private double prix;

    @ManyToOne
    @JoinColumn(name = "category_id")
    @JsonIgnoreProperties("produits") 
    private Category category;

    // Getters et setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }

    public double getPrix() { return prix; }
    public void setPrix(double prix) { this.prix = prix; }

    public Category getCategory() { return category; }
    public void setCategory(Category category) { this.category = category; }
}